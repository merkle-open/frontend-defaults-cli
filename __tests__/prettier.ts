/// <reference types="@types/jest" />
import { apiIt } from './shared';

describe('prettier', () => {
	it('default', async () =>
		await apiIt('prettier', {
			prettier: true,
		}));
	it('with-githooks', async () =>
		await apiIt('prettier-with-githooks', {
			prettier: true,
			githooks: true,
		}));
});
