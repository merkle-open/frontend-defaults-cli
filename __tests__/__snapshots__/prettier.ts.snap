// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettier default 1`] = `"{  \\"options\\": {    \\"ts\\": false,    \\"tslint\\": false,    \\"es\\": false,    \\"eslint\\": false,    \\"editorconfig\\": false,    \\"prettier\\": true,    \\"stylelint\\": false,    \\"gitignore\\": false,    \\"npmrc\\": false,    \\"readme\\": false,    \\"githooks\\": false,    \\"commitlint\\": false,    \\"nodeVersion\\": false,    \\"webpack\\": false,    \\"install\\": false,    \\"force\\": true,    \\"dryRun\\": false,    \\"mode\\": \\"api\\"  },  \\"mergedFiles\\": {    \\"package.json\\": {      \\"override\\": false,      \\"data\\": {        \\"scripts\\": {          \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"        },        \\"devDependencies\\": {          \\"@namics/prettier-config\\": \\"0.2.0\\",          \\"prettier\\": \\"1.16.3\\"        }      }    },    \\".prettierignore\\": {      \\"override\\": false,      \\"data\\": \\"# Node\\\\n**/node_modules\\\\n**/package.json\\\\n**/package-lock.json\\\\n\\\\n# Generated\\\\n**/bower.json\\\\n**/lerna.json\\\\n\\\\n# Build\\\\n**/build\\\\n**/dist\\\\n**/public\\\\n**/coverage\\\\n**/storybook-static\\\\n\\\\n# Nitro\\\\n**/project/blueprints\\\\n**/.yo-rc.json\\"    },    \\".prettierrc.js\\": {      \\"override\\": false,      \\"data\\": \\"module.exports = require('@namics/prettier-config');\\\\n\\"    }  }}"`;

exports[`prettier default 2`] = `"# Node**/node_modules**/package.json**/package-lock.json# Generated**/bower.json**/lerna.json# Build**/build**/dist**/public**/coverage**/storybook-static# Nitro**/project/blueprints**/.yo-rc.json"`;

exports[`prettier default 3`] = `"module.exports = require('@namics/prettier-config');"`;

exports[`prettier default 4`] = `"{  \\"scripts\\": {    \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"  },  \\"devDependencies\\": {    \\"@namics/prettier-config\\": \\"0.2.0\\",    \\"prettier\\": \\"1.16.3\\"  }}"`;

exports[`prettier with-githooks 1`] = `"{  \\"options\\": {    \\"ts\\": false,    \\"tslint\\": false,    \\"es\\": false,    \\"eslint\\": false,    \\"editorconfig\\": false,    \\"prettier\\": true,    \\"stylelint\\": false,    \\"gitignore\\": false,    \\"npmrc\\": false,    \\"readme\\": false,    \\"githooks\\": true,    \\"commitlint\\": false,    \\"nodeVersion\\": false,    \\"webpack\\": false,    \\"install\\": false,    \\"force\\": true,    \\"dryRun\\": false,    \\"mode\\": \\"api\\"  },  \\"mergedFiles\\": {    \\"package.json\\": {      \\"override\\": false,      \\"data\\": {        \\"lint-staged\\": {          \\"linters\\": {            \\"*.{js,jsx,ts,tsx,json}\\": [              \\"prettier --list-different --write\\"            ]          }        },        \\"scripts\\": {          \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"        },        \\"husky\\": {          \\"hooks\\": {            \\"pre-commit\\": \\"lint-staged\\"          }        },        \\"devDependencies\\": {          \\"@namics/prettier-config\\": \\"0.2.0\\",          \\"husky\\": \\"1.3.1\\",          \\"lint-staged\\": \\"8.1.1\\",          \\"prettier\\": \\"1.16.3\\"        }      }    },    \\".huskyrc\\": {      \\"override\\": false,      \\"data\\": \\"# ~/.huskyrc\\\\n# Note: it will run in /bin/sh shell\\\\nif command -v nodenv >/dev/null 2>&1; then\\\\n  eval \\\\\\"$(nodenv init -)\\\\\\"\\\\nfi\\"    },    \\".prettierignore\\": {      \\"override\\": false,      \\"data\\": \\"# Node\\\\n**/node_modules\\\\n**/package.json\\\\n**/package-lock.json\\\\n\\\\n# Generated\\\\n**/bower.json\\\\n**/lerna.json\\\\n\\\\n# Build\\\\n**/build\\\\n**/dist\\\\n**/public\\\\n**/coverage\\\\n**/storybook-static\\\\n\\\\n# Nitro\\\\n**/project/blueprints\\\\n**/.yo-rc.json\\"    },    \\".prettierrc.js\\": {      \\"override\\": false,      \\"data\\": \\"module.exports = require('@namics/prettier-config');\\\\n\\"    }  }}"`;

exports[`prettier with-githooks 2`] = `"# ~/.huskyrc# Note: it will run in /bin/sh shellif command -v nodenv >/dev/null 2>&1; then  eval \\"$(nodenv init -)\\"fi"`;

exports[`prettier with-githooks 3`] = `"# Node**/node_modules**/package.json**/package-lock.json# Generated**/bower.json**/lerna.json# Build**/build**/dist**/public**/coverage**/storybook-static# Nitro**/project/blueprints**/.yo-rc.json"`;

exports[`prettier with-githooks 4`] = `"module.exports = require('@namics/prettier-config');"`;

exports[`prettier with-githooks 5`] = `"{  \\"lint-staged\\": {    \\"linters\\": {      \\"*.{js,jsx,ts,tsx,json}\\": [        \\"prettier --list-different --write\\"      ]    }  },  \\"scripts\\": {    \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"  },  \\"husky\\": {    \\"hooks\\": {      \\"pre-commit\\": \\"lint-staged\\"    }  },  \\"devDependencies\\": {    \\"@namics/prettier-config\\": \\"0.2.0\\",    \\"husky\\": \\"1.3.1\\",    \\"lint-staged\\": \\"8.1.1\\",    \\"prettier\\": \\"1.16.3\\"  }}"`;
